subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/SyncTaskIntegrationTest.groovy

```diff
@@ -394,36 +393,50 @@ class SyncTaskIntegrationTest extends AbstractIntegrationSpec {
         ins.close()
     }
 
-    @ToBeImplemented
     @Issue("https://github.com/gradle/gradle/issues/9586")
-    @Requires(TestPrecondition.CASE_INSENSITIVE_FS)
     def "change in case of input file still syncs properly"() {
         given:
         buildFile << """
-            task writeFile {
-                def outputFile = new File(buildDir, project.hasProperty("capitalize") ? "OUTPUT" : "output")
-                outputs.file outputFile
-                doLast {
-                    outputFile.text = "hello"
-                }
-            }
             task sync(type: Sync) {
-                from writeFile
-                into new File(buildDir, "sync")
+                from project.hasProperty("capitalize") ? "FILE.TXT" : "file.txt"
+                from "dir"
+                into buildDir
             }
         """
-        when:
-        succeeds("sync")
-        then:
-        file("build/output").assertExists()
-        file("build/sync/output").assertExists()
+
+        def lowercaseFile = file("file.txt")
+        lowercaseFile.text = "lowercase"
+
+        file("dir").mkdirs()
+        file("dir/file2.txt").createNewFile()
+        file("dir/file3.txt").createNewFile()
+
+        and:
+        run "sync"
+        file("build/file.txt").assertExists()
+        file("build/file2.txt").assertExists()
+        file("build/file3.txt").assertExists()
+
+        and:
+        lowercaseFile.delete() // renameTo wouldn't change the case
+        def uppercaseFile = file("FILE.TXT")
```

Comment:
I think we should uppercase a file in the directory as well.

Reply:
Done, thanks